python/orca/src/bigdl/orca/common.py

```diff
@@ -206,120 +207,130 @@ def init_orca_context(cluster_mode=None, cores=2, memory="2g", num_nodes=1,
     print("Initializing orca context")
     import atexit
     atexit.register(stop_orca_context)
-    from pyspark import SparkContext
-    import warnings
-    spark_args = {}
-    for key in ["conf", "spark_log_level", "redirect_spark_log"]:
-        if key in kwargs:
-            spark_args[key] = kwargs[key]
-    if cluster_mode is not None:
-        cluster_mode = cluster_mode.lower()
-    activate_sc = SparkContext._active_spark_context is not None
-    if activate_sc:
-        if cluster_mode is not None and cluster_mode != "spark-submit":
-            warnings.warn("Use an existing SparkContext, " +
-                          "cluster_mode is determined by the existing SparkContext", Warning)
-        from bigdl.dllib.nncontext import init_nncontext
-        sc = init_nncontext(conf=None, spark_log_level="WARN", redirect_spark_log=True)
-    else:
-        cluster_mode = "local" if cluster_mode is None else cluster_mode
-        if cluster_mode == "local":
-            if num_nodes > 1:
-                warnings.warn("For Spark local mode, num_nodes should be 1, but got "
-                              + repr(num_nodes) + ", ignored", Warning)
-            os.environ["SPARK_DRIVER_MEMORY"] = memory
-            if "python_location" in kwargs:
-                spark_args["python_location"] = kwargs["python_location"]
-            from bigdl.dllib.nncontext import init_spark_on_local
-            sc = init_spark_on_local(cores, **spark_args)
-        elif cluster_mode == "spark-submit":
+    if runtime == "ray":
+        assert cluster_mode is None, "cluster mode should be None"
```

Comment:
Maybe a more clear message? "Currently, cluster mode is not supported for ray runtime and you must connect to an exiting ray cluster."